src/screens/CurrentTrack/index.tsx

```diff
@@ -35,66 +38,52 @@ const parseAndPostWhoSampledData = async (artists: ArtistObjectSimplified[], nam
     return false;
   }
 };
-
-const useBedfellowService = () => {
-  const [isLoading, setIsLoading] = useState<boolean>(false);
-  const [trackSamples, setTrackSamples] = useState<BedfellowTrackSamples | null>(null);
-  const [refresh, setRefresh] = useState<boolean>(false);
-
-  const loadBedfellowData = async (artists: ArtistObjectSimplified[], track: string) => {
-    setIsLoading(true);
-
-    try {
-      if ((artists[0].name, track)) {
-        const trackSamplesResults = await Promise.all(
-          await artists.map(async (artist) => {
-            const result: BedfellowTrackSamples | null = await getBedfellowDBData(artist?.name, track);
-            return result;
-          })
-        );
-        // This could lead to a situation where we make a bunch of calls then only use the first result
-        // let's try to make this smarter later
-        const filteredTrackSampleResults = trackSamplesResults.filter((trackSample) => trackSample);
-        if (filteredTrackSampleResults.length) {
-          setTrackSamples(filteredTrackSampleResults[0]);
-          setIsLoading(true);
-          return;
-        }
-      }
-      const result = await parseAndPostWhoSampledData(artists, track);
-      if (result) {
-        const sampleData = await getBedfellowDBData(artists[0].name, track);
-        setTrackSamples(sampleData);
+const loadBedfellowData = async (artists: ArtistObjectSimplified[] = [], track: string = '') => {
+  try {
+    if ((artists[0].name, track)) {
+      const trackSamplesResults = await Promise.all(
+        await artists.map(async (artist) => {
+          const result: BedfellowTrackSamples | null = await getBedfellowDBData(artist?.name, track);
+          return result;
+        })
+      );
+      // This could lead to a situation where we make a bunch of calls then only use the first result
+      // let's try to make this smarter later
+      const filteredTrackSampleResults = trackSamplesResults.filter((trackSample) => trackSample);
+      if (filteredTrackSampleResults.length) {
+        return filteredTrackSampleResults[0];
       }
-
-      setIsLoading(false);
-    } catch (error) {
-      console.log('ERROR::', error);
-      setIsLoading(false);
     }
-  };
-
-  const refreshTrackSamples = () => {
-    setRefresh(!refresh);
-    setTrackSamples(null);
-  };
-
-  return { isLoading, trackSamples, loadBedfellowData, refreshTrackSamples };
+    const result = await parseAndPostWhoSampledData(artists, track);
+    if (result) {
+      return await getBedfellowDBData(artists[0].name, track);
+    }
+  } catch (error) {
+    console.log('loadBedfellowData error::', error);
+    return null;
+  }
 };
 
 export function CurrentSong({ item }: CurrentSongProps) {
+  const colors: IOSImageColors | AndroidImageColors | null = useImagePalette(item?.album?.images?.[0].url || '');
+  const backgroundStyle: ViewStyle = {
```

Comment:
Move these to the styles file

Reply:
Move the hook for using the image palette out and into the parent component, then just use the context in here.

Reply:
Done